spectec/src/util/lib.ml

```diff
@@ -11,6 +11,14 @@ struct
     | x::xs -> let ys, y = split_last xs in x::ys, y
     | [] -> failwith "split_last"
 
+  let split_last_opt = function
+    | x::[] -> [], Some x
+    | x::xs -> let ys, y = split_last xs in x::ys, Some y
+    | [] -> [], None
```

Comment:
Nit: to avoid code duplication, perhaps reimplement split_last in terms of split_last_opt now?